commands/music/play.js

```diff
@@ -222,9 +222,8 @@ module.exports = class PlayCommand extends Command {
         return message.reply(
           'The queue is full, please try adding more songs later'
         );
-      videosArr
-        .splice(0, maxQueueLength - message.guild.musicData.queue.length)
-        .forEach(async (video, key, arr) => {
+      videosArr = videosArr.splice(0, maxQueueLength - message.guild.musicData.queue.length);
+        await videosArr.reduce(async (memo,video, key, arr) => {
           // don't process private videos
```

Comment:
you saw that `videosArr =` was missing too, but what is with `memo`?

and the `, undefined` on line 262

Reply:
The way that the reduce command works is that it receives a callback function (that receives 4 arguments, an accumulator, the current value, the index of the current value in the array and the array itself. The accumulator basically adds the return value of the previous callback function.) and an initial value(optional, basically accumulator = initial value if it exists, otherwise 0).
reduce activates the callback function on each of the array elements. The memo is the accumulator and from what I understand, because There  is an await on it , it should wait for the previous callback to return and so the videos will be pushed to the queue in order. The undefined just doesn't give the reduce function an initial value.

Reply:
thank you so much for the explanation <3

i hav't seen that before and peeked my curiosity

Much Love
-Bacon